src/java.base/share/classes/jdk/internal/util/ClassFileDumper.java

```diff
@@ -167,48 +129,62 @@ public void dumpFailedClass(String name, byte[] bytes) {
     }
 
     @SuppressWarnings("removal")
-    private void write(Path path, byte[] bytes) {
+    private void write(File file, byte[] bytes) {
+        // use java.io.File instead of java.nio.file.Path to avoid
+        // recursive initialization during early startup. e.g.
+        // Enum::valueOf causes method handle to be invoked which
+        // may cause classes to be spinned and dumped
```

Comment:
This comment could be omitted; it won't age well and is out of context for the fix.